python/ray/serve/tests/test_deploy.py

```diff
@@ -340,17 +346,36 @@ async def __call__(self):
             return self.state["a"]
 
     handle = serve.run(A.options(version="1", user_config={"a": 1}).bind())
-    responses = [handle.remote() for _ in range(30)]
+    responses = [handle.remote() for _ in range(10)]
+
+    # Give the queries time to get to the replicas before the reconfigure.
+    time.sleep(0.1)
```

Comment:
instead of sleep you can check `signal.get_cur_num_waiters` to explicitly wait for them to start running (not 100% sure about method name you can search)

Reply:
`time.sleep` should (almost) never be used in the tests else it'll inevitably end up flaky

Reply:
Good point! Didn't know that method was available, lemme give it a try.